##### Measures of Central Tendency:

1) Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.
2) Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.
3) Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.
4) Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.


In [106]:
import numpy as np

In [107]:
file_path =(r"C:\Users\admin\Desktop\JUPYTER\pandas\Sample_data.csv")

In [100]:
data = np.genfromtxt(file_path, delimiter=',',skip_header=1 )
print(data)

[[ 0. 25. 40.]
 [ 1. 30. 35.]
 [ 2. 28. 38.]
 [ 3. 32. 41.]
 [ 4. 26. 37.]
 [ 5. 29. 36.]
 [ 6. 27. 39.]
 [ 7. 33. 42.]
 [ 8. 31. 38.]
 [ 9. 28. 37.]
 [10. 30. 40.]
 [11. 29. 36.]]


1) Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.

In [85]:
data.shape

(13, 3)

In [94]:
product_A = data[:,1]
print(product_A)
mean_product_A = np.mean(product_A)
print("Mean of Product A is : ",mean_product_A)

[25. 30. 28. 32. 26. 29. 27. 33. 31. 28. 30. 29.]
Mean of Product A is :  29.0


In [95]:
product_B = data[:,2]
print(product_B)
mean_product_B = np.mean(product_B)
print("Mean of Product B is : ",mean_product_B)

[40. 35. 38. 41. 37. 36. 39. 42. 38. 37. 40. 36.]
Mean of Product B is :  38.25


 Mean (Average):
* ProductA has an average monthly sales of 29 units.
* ProductB has an average monthly sales of 38.25 units.

Comparison:
* ProductB has a higher average monthly sales compared to ProductA.

In [119]:
#Calculating percentiles
product_A_25 = np.percentile(product_A,25)
product_A_50= np.percentile(product_A,50)
product_A_75= np.percentile(product_A,75)
print("25% : ",product_A_25)
print("50% : " , product_A_50)
print("75% : ", product_A_75)

25% :  27.75
50% :  29.0
75% :  30.25


In [118]:
product_B_25 = np.percentile(product_B,25)
product_B_50= np.percentile(product_B,50)
product_B_75= np.percentile(product_B,75)
print("25% : ",product_B_25)
print("50% : " , product_B_50)
print("75% : ", product_B_75)

25% :  36.75
50% :  38.0
75% :  40.0


* Looking at the percentiles (25%, 50%, 75%), we can see that ProductB consistently has higher sales values than ProductA. This reinforces the observation that ProductB generally outperforms ProductA across different sales levels.


2) Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.

In [126]:
median_A = np.median(product_A)
print("median sales of product A : ",median_A)

median sales of product A :  29.0


In [127]:
median_B = np.median(product_B)
print("median sales of product B : ",median_B)

median sales of product B :  38.0





    median of product B is middle value it indicates that below median sales are low and above medium sales are high

    median shows a clear middle point in the sales data for both products



3) Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.

In [130]:
import statistics as sts

In [131]:
sts.mode(product_A)

30.0

In [132]:
sts.mode(product_B)

40.0

4) Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.

![](https://www.gstatic.com/education/formulas2/553212783/en/weighted_average_formula.svg)
W	=	weighted average

n	=	number of terms to be averaged

w_{i}	=	weights applied to x values

X_{i}	=	data values to be averaged


In [141]:
weighted_mean_A = np.mean(product_A)  # each month has an equal weight, the weighted mean is equivalent to the regular mean.
print("Weighted mean of product A : ",weighted_mean_A)  
weighted_mean_B = np.mean(product_B)
print("Weighted mean of product B : ",weighted_mean_B)

Weighted mean of product A :  29.0
Weighted mean of product B :  38.25


###### Measures of Dispersion:
1. Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.


In [146]:
range_product_A=np.max(product_A)-np.min(product_A)
print("Range of Product A : ",range_product_A)

Range of Product A :  8.0


In [147]:
range_product_B=np.max(product_B)-np.min(product_B)
print("Range of Product B : ",range_product_B)

Range of Product B :  7.0


2) Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.

In [148]:
Variance_A = np.var(product_A)
print(Variance_A)

5.166666666666667


In [149]:
Variance_B = np.var(product_B)
print(Variance_B)

4.354166666666667


3) Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.

In [152]:
# Standard Deviation  
std_deviation_productA = np.std(product_A)
print(std_deviation_productA)
std_deviation_productB = np.std(product_B)
print(std_deviation_productB)

2.273030282830976
2.0866640042581523


###### Relative Measures of Dispersion:
1.Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.


In [153]:
print(product_A)

[25. 30. 28. 32. 26. 29. 27. 33. 31. 28. 30. 29.]


In [163]:
print(mean_product_A)
print(std_deviation_productA)
c = (std_deviation_productA/mean_product_A)*100
print("coefficient of variation for the sales data of Product A : ",c)

29.0
2.273030282830976
coefficient of variation for the sales data of Product A :  7.838035458037848


2. Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.

In [165]:
range_product_B

7.0

In [170]:
relative_range_B = (range_product_B/mean_product_B)*100
print(relative_range_B)

18.30065359477124


3. Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.



In [178]:
std_A=np.std(product_A)
mean_product_A
relative_std_A = (std_A/mean_product_A)*100
print(relative_std_A)

7.838035458037848




    Product A mean is 29.0 and relative standard deviation of product A is 7.838035458037848
    Product A is with low CV values have less relative variability around their mean, sales are stable consistent

